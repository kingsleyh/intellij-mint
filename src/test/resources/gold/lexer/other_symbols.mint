enum Cars {
  Mercedes
  Volvo
  BMW
}

component A {

  state error : String = ""
  property name : String = ""

  fun js() : Thing {
    `var name = 'cool' ; return name;`
  }

  fun a() : Void {
    sequence {
      if(a == 1 && b != 2 || c <= 3 || c > a && b < x || b >= y){
        true
      } else {
        false
      }

      a = !true

      b = 1 + 1 - 4 * 8 / 6 % 2

      c = Cars::Volvo

      someList
      |> Array.map(getName)
      |> Array.filter((n : String) : String => { n + "_hello"})

    } catch Error => error {
      void
    } catch String => error {
      next { error = error}
    }

  } where {
    myList
  }

}
