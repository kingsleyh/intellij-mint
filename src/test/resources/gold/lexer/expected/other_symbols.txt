enum ('enum')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Cars')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n  ')
CAPITALISED_IDENT ('Mercedes')
WHITE_SPACE ('\n  ')
CAPITALISED_IDENT ('Volvo')
WHITE_SPACE ('\n  ')
CAPITALISED_IDENT ('BMW')
WHITE_SPACE ('\n')
} ('}')
WHITE_SPACE ('\n\n')
component ('component')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('A')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n\n  ')
state ('state')
WHITE_SPACE (' ')
LOWER_IDENT ('error')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
STRING_LITERAL ('""')
WHITE_SPACE ('\n  ')
property ('property')
WHITE_SPACE (' ')
LOWER_IDENT ('name')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
STRING_LITERAL ('""')
WHITE_SPACE ('\n\n  ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('js')
( ('(')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Thing')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
STRING_LITERAL ('`var name = 'cool' ; return name;`')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n  ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('a')
( ('(')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Void')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
sequence ('sequence')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
if ('if')
( ('(')
LOWER_IDENT ('a')
WHITE_SPACE (' ')
== ('==')
WHITE_SPACE (' ')
NUMBER_LITERAL ('1')
WHITE_SPACE (' ')
&& ('&&')
WHITE_SPACE (' ')
LOWER_IDENT ('b')
WHITE_SPACE (' ')
!= ('!=')
WHITE_SPACE (' ')
NUMBER_LITERAL ('2')
WHITE_SPACE (' ')
|| ('||')
WHITE_SPACE (' ')
LOWER_IDENT ('c')
WHITE_SPACE (' ')
<= ('<=')
WHITE_SPACE (' ')
NUMBER_LITERAL ('3')
WHITE_SPACE (' ')
|| ('||')
WHITE_SPACE (' ')
LOWER_IDENT ('c')
WHITE_SPACE (' ')
> ('>')
WHITE_SPACE (' ')
LOWER_IDENT ('a')
WHITE_SPACE (' ')
&& ('&&')
WHITE_SPACE (' ')
LOWER_IDENT ('b')
WHITE_SPACE (' ')
< ('<')
WHITE_SPACE (' ')
LOWER_IDENT ('x')
WHITE_SPACE (' ')
|| ('||')
WHITE_SPACE (' ')
LOWER_IDENT ('b')
WHITE_SPACE (' ')
>= ('>=')
WHITE_SPACE (' ')
LOWER_IDENT ('y')
) (')')
{ ('{')
WHITE_SPACE ('\n        ')
true ('true')
WHITE_SPACE ('\n      ')
} ('}')
WHITE_SPACE (' ')
else ('else')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n        ')
false ('false')
WHITE_SPACE ('\n      ')
} ('}')
WHITE_SPACE ('\n\n      ')
LOWER_IDENT ('a')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
! ('!')
true ('true')
WHITE_SPACE ('\n\n      ')
LOWER_IDENT ('b')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
NUMBER_LITERAL ('1')
WHITE_SPACE (' ')
+ ('+')
WHITE_SPACE (' ')
NUMBER_LITERAL ('1')
WHITE_SPACE (' ')
- ('-')
WHITE_SPACE (' ')
NUMBER_LITERAL ('4')
WHITE_SPACE (' ')
* ('*')
WHITE_SPACE (' ')
NUMBER_LITERAL ('8')
WHITE_SPACE (' ')
/ ('/')
WHITE_SPACE (' ')
NUMBER_LITERAL ('6')
WHITE_SPACE (' ')
% ('%')
WHITE_SPACE (' ')
NUMBER_LITERAL ('2')
WHITE_SPACE ('\n\n      ')
LOWER_IDENT ('c')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Cars')
:: ('::')
CAPITALISED_IDENT ('Volvo')
WHITE_SPACE ('\n\n      ')
LOWER_IDENT ('someList')
WHITE_SPACE ('\n      ')
|> ('|>')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Array')
. ('.')
LOWER_IDENT ('map')
( ('(')
LOWER_IDENT ('getName')
) (')')
WHITE_SPACE ('\n      ')
|> ('|>')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Array')
. ('.')
LOWER_IDENT ('filter')
( ('(')
( ('(')
LOWER_IDENT ('n')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE (' ')
LOWER_IDENT ('n')
WHITE_SPACE (' ')
+ ('+')
WHITE_SPACE (' ')
STRING_LITERAL ('"_hello"')
} ('}')
) (')')
WHITE_SPACE ('\n\n    ')
} ('}')
WHITE_SPACE (' ')
catch ('catch')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Error')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('error')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
void ('void')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE (' ')
catch ('catch')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('String')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('error')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
next ('next')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE (' ')
LOWER_IDENT ('error')
WHITE_SPACE (' ')
= ('=')
WHITE_SPACE (' ')
LOWER_IDENT ('error')
} ('}')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n  ')
} ('}')
WHITE_SPACE (' ')
where ('where')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
LOWER_IDENT ('myList')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n')
} ('}')